Sources/SwiftKuery/Connection.swift

```diff
@@ -14,64 +14,111 @@
  limitations under the License.
  */
 
+import Dispatch
+
 // MARK: Connection protocol
 
 /// Defines the protocol which all database plugins must implement.
-public protocol Connection {
-  
+public protocol Connection : class {
+
     /// The `QueryBuilder` with connection specific substitutions.
     var queryBuilder: QueryBuilder { get }
     
     /// Establish a connection with the database.
     ///
     /// - Parameter onCompletion: The function to be called when the connection is established.
-    func connect(onCompletion: (QueryError?) -> ())
+    func connect(onCompletion: @escaping (QueryError?) -> ())
+
+    /// Establish a connection with the database.
+    ///
+    /// - Returns: A QueryError if the connection cannot connect, otherwise nil
+    func connectSync() -> QueryError?
 
     /// Close the connection to the database.
     func closeConnection()
-    
+
     /// An indication whether there is a connection to the database.
     var isConnected: Bool { get }
-    
+
     /// Execute a query.
     ///
     /// - Parameter query: The query to execute.
     /// - Parameter onCompletion: The function to be called when the execution of the query has completed.
     func execute(query: Query, onCompletion: @escaping ((QueryResult) -> ()))
+
+    /// Execute a query.
+    ///
+    /// - Parameter query: The query to execute.
+    /// - Returns: A QueryResult representing the result of the execute operation
+    func executeSync(query: Query) -> QueryResult
     
     /// Execute a raw query.
     ///
-    /// - Parameter query: A String with the query to execute.
+    /// - Parameter raw: A String with the query to execute.
```

Comment:
These doc comments are for external parameter names, so this should be "`Parameter _:`". They'll need fixing up elsewhere too.

Reply:
I have looked for precedent on how to document parameters with no external name. Xcode will generate documentation which identifies the parameter using its internal name so I suggest we leave these as they are to be inline with that behavior.